presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -47,18 +48,21 @@ public QueryResource(QueryManager queryManager)
     }
 
     @GET
-    public List<BasicQueryInfo> getAllQueryInfo()
+    public List<BasicQueryInfo> getAllQueryInfo(@QueryParam("state") String queryState)
     {
-        return extractBasicQueryInfo(queryManager.getAllQueryInfo());
+        return queryManager.getAllQueryInfo()
+                .stream()
+                .filter(queryInfo -> includeQueryInfo(queryInfo, queryState))
```

Comment:
maybe just inline a `queryInfo -> queryState == null || queryState.equalsIgnoreCase(queryInfo.getState().toString())`... Alternatively you could convert the queryState string to an enum value up front and then just use `==`.